client.go

```diff
@@ -784,9 +802,17 @@ func (c *Client) establishRegion(originalReg *regioninfo.Info, host string, port
 			}
 		}
 		if c.clientType == AdminClient {
-			host, port, err = c.zkLookup(ctx, zk.Master)
+			if c.master == (zk.ResourceName)("") {
```

Comment:
You could initialize `c.master` and `c.meta` in the constructor (https://github.com/assumednormal/gohbase/blob/master/client.go#L226) if they are empty.